src/otx/algorithms/anomaly/tasks/inference.py

```diff
@@ -313,6 +317,32 @@ def export(
         output_model.set_data("label_schema.json", label_schema_to_bytes(self.task_environment.label_schema))
         self._set_metadata(output_model)
 
+    def _convert_to_new_format(self, xml_file: str) -> None:
+        """Adds the metadata to the model IR.
+
+        Adds the metadata to the model IR. So that it can be used with the new modelAPI.
+        This is because the metadata.json is not used by the new modelAPI.
+        # TODO CVS-114640
+        # TODO: Step 1. Remove metadata.json when modelAPI becomes the default inference method.
+        # TODO: Step 2. Remove this function when Anomalib is upgraded as the model graph will contain the required ops
+        # TODO: Step 3. Update modelAPI to remove pre/post-processing steps when Anomalib version is upgraded.
+        """
+        metadata = self._get_metadata_dict()
+        core = Core()
+        model = core.read_model(xml_file)
+        for key, value in metadata.items():
+            if key == "transform":
+                continue
+            model.set_rt_info(value, ["model_info", key])
+        # Add transforms
+        if "transform" in metadata:
+            for transform_dict in metadata["transform"]["transform"]["transforms"]:
+                transform = transform_dict.pop("__class_fullname__")
+                for key, value in transform_dict.items():
+                    model.set_rt_info(value, ["model_info", "transforms", transform, key])
+        model.set_rt_info("AnomalyDetection", ["model_info", "model_type"])
+        serialize(model, xml_file)
```

Comment:
In OV 2023 we can't read and then serialize to the same file: possible workaround is here https://github.com/sovrasov/openvino_training_extensions/blob/vs/upd_ov/src/otx/algorithms/common/utils/ir.py#L36